include/nbl/builtin/hlsl/workgroup2/arithmetic_config.hlsl

```diff
@@ -75,6 +72,16 @@ struct ArithmeticConfiguration
     static_assert(ItemsPerInvocation_1<=4, "3 level scan would have been needed with this config!");
 
     NBL_CONSTEXPR_STATIC_INLINE uint16_t ElementCount = conditional_value<LevelCount==1,uint16_t,0,conditional_value<LevelCount==3,uint16_t,SubgroupSize*ItemsPerInvocation_2,0>::value + SubgroupSize*ItemsPerInvocation_1>::value;
+
+    static uint32_t virtualSubgroupID(const uint32_t id, const uint32_t offset)
+    {
+        return offset * (WorkgroupSize >> SubgroupSizeLog2) + id;
+    }
+
+    static uint32_t sharedMemCoalescedIndex(const uint32_t id, const uint32_t itemsPerInvocation)
+    {
+        return (id & (itemsPerInvocation-1)) * SubgroupSize + (id/itemsPerInvocation);
```

Comment:
again really bad name, I have no idea what `id` is by reading the code, btw you could probably as for `itemsPerInvocation` as a template and a Log2 at that

Also you're multiplying by `SubgroupSize` not `SubgroupsPerVirtualWorkgroup` (so baiscally Virtual Subgroup Count) as in the original code.... did you name things wrong again ?